homeassistant/components/plex/server.py

```diff
@@ -78,13 +89,73 @@ def _connect_with_url():
 
         _connect_with_url()
 
-    def clients(self):
-        """Pass through clients call to plexapi."""
-        return self._plex_server.clients()
-
-    def sessions(self):
-        """Pass through sessions call to plexapi."""
-        return self._plex_server.sessions()
+    def refresh_entity(self, machine_identifier, device, session):
+        """Forward refresh dispatch to media_player."""
+        dispatcher_send(
+            self._hass,
+            PLEX_UPDATE_MEDIA_PLAYER_SIGNAL.format(machine_identifier),
+            device,
+            session,
+        )
+
+    def update_platforms(self):
+        """Update the platform entities."""
+        available_clients = {}
+        new_clients = []
+
+        try:
+            devices = self._plex_server.clients()
+            sessions = self._plex_server.sessions()
+        except plexapi.exceptions.BadRequest:
+            _LOGGER.exception("Error requesting Plex client data from server")
+            return
+        except requests.exceptions.RequestException as ex:
+            _LOGGER.warning(
+                "Could not connect to Plex server: %s (%s)", self.friendly_name, ex
+            )
+            return
+
+        for device in devices:
+            available_clients[device.machineIdentifier] = {"device": device}
+
+            if device.machineIdentifier not in self._known_clients:
```

Comment:
If we don't need the debug log we can simplify to:
```suggestion
            new_clients.add(device.machineIdentifier)
```

Reply:
We only want to add the ID to `new_clients` if it doesn't already exist in `self._known_clients`, so this check is still necessary.

I'd like to keep debug log in case it's needed to troubleshoot after this logic overhaul.

Reply:
Adding to a set is a no op if the item already exist in the set.

Reply:
I see I didn't read carefully. We're checking a different container than we're updating.